esy-build-package/EsyBash.re

```diff
@@ -0,0 +1,77 @@
+module System = EsyLib.System;
+
+/** 
+ * Helper method to get the root path of the 'esy-bash' node modules
+ */
+let getEsyBashRootPath = () => {
+  open Run;
+  let program = Sys.argv[0];
+  let%bind program = realpath(v(program));
+  let basedir = Fpath.parent(program);
+  let resolution =
+      EsyLib.NodeResolution.resolve(
+        "../../../../node_modules/esy-bash/package.json",
+        basedir,
+      );
+
+  switch%bind (Run.coerceFrmMsgOnly(resolution)) {
+    | Some(path) => Ok(Fpath.parent(path))
+    | None => Error(`Msg("Unable to find 'esy-bash'"))
+  };
+};
+
+/**
+ * Helper method to get the `cygpath` utility path
+ * Used for resolving paths
+ */
+let getCygPath = () => {
+    open Run;
+    let%bind rootPath = getEsyBashRootPath();
+    Ok(Fpath.(rootPath / ".cygwin" / "bin" / "cygpath.exe"));
+};
+
+let getEsyBashPath = () => {
+    open Run;
+    let%bind rootPath = getEsyBashRootPath();
+    Ok(Fpath.(rootPath / "bin" / "esy-bash.js"));
+};
+
+/**
+* Helper utility to normalize paths to a cygwin style,
+* ie, "C:\temp" -> "/cygdrive/c/temp"
+* On non-Windows platforms, this is a noop
+*/
+let normalizePathForCygwin = (path) => {
+    open Run;
+    switch (System.host) {
+        | System.Windows => {
+            let%bind rootPath = getCygPath();
+            let ic = Unix.open_process_in(Fpath.to_string(rootPath) ++ " \"" ++ path ++ " \"")
```

Comment:
@andreypopp , let me know if there is anything you'd like me to restructure or change here - first real `reason` code I've written, so I'm sure I did a lot of things wrong.